### ISSUE TITLE
Early Detection of Sexual Predators in Chats | Matthias Vogt Ulf Leser Alan Akbik

### ISSUE THREAD

**Issue by hamedwaezi01** [fani-lab/Osprey#32]
**Early Detection of Sexual Predators in Chats
Matthias Vogt Ulf Leser Alan Akbik
August 2021**
https://aclanthology.org/2021.acl-long.386/

**Why I chose this paper?**
It is one of the recent papers that worked on the early detection of predatory. It also suggested a new dataset which was derived from PAN-12 and ChatCoder2.
This paper implemented the code and linked a Git repository, so reproduction is less of a problem.

**The main problem:**
Online predatory is a problem that researchers have been working on. There are two problems in this subject. Initially, detecting the predator is usually done after seeing the whole predatory conversation and may not be able to detect it before the predator abuses the minor.
Moreover, the current widely used dataset is PAN-12. Its main problem is the trimming of conversations based on time and length of them. It means a lengthy predatory chat is cut into several separate conversations, although they are of the same chat. So different stages of a predatory chat are present in different dataset records. As a result, we might not rely on such a dataset.

**Applications:**
They try detecting sexual predator detection early in conversations. Although they state the work is early detection of predators, they try to detect the predatory conversation, not the predator themselves.

**Existing Works:**
- Datasets:
PAN-12 is the dataset for PAN 2012 competition, the predatory task.
VTPAN is derived from PAN-12 by removing some of the records.
ChatCoder 2 (CC2) contains 497 complete grooming chats from PJ
- Methods:
Early detection of deception and aggressiveness using profile-based representations Escalante et al. (2017)
**Early Text Classification using Multi-Resolution Concept Representations, Monroy et al. (2018) SOTA**
Bours and Kulsrud (2019) also studied this problem, but it was not extensive.

**Method:**
The eSPD system has two stages. 1) a window of (50) messages that classifies them. 2) A predictor that decides whether raise an alert or not.
The first stage uses a pre-trained transformer model (BERT, e.g.) followed by a linear classifier. Then “fine-tune the entire architecture”. The output is a binary prediction. They used three variants of BERT, including BERT-base, BERT-large, and MobileBERT.
The second stage looks at a range of windows, 10 in this paper, and uses a threshold (skepticism) to label the chat predatory. So, for skepticism of 4, if we have five windows labeled positive, it gets classified as grooming.

**Input and Output:**
A set of chats consists of a series of messages where “a message is a string with a time and an author”. It is not clear whether the model includes the time in the input.
The system's output determines whether the state of a chat until the current message is predatory.

**Gaps:**
I believe their dataset needs better negative samples. Theirs are the PAN-12 which are trimmed conversations. My argument is that the model might learn that a lengthy chat is always predatory because a non-predatory chat has a cap.
The authors state that they fine-tuned the model in the first stage. It sounds very ambiguous how they fine-tuned the transformer model. Did they first fine-tune the transformer with the dataset input and then pass the embedding to the linear classifier? Or did they give the string to the transformer, then to the linear layer, and afterward backpropagated?

**Results:**
They tested the first stage of the system against Escalante et al. (2017) and Pastor Lopez-Monroy et al. (2018). They fed fractions of (from zero to 100%) the VTPAN to the models. The F1 scores of BERT-base and BERT-large were significantly better than that of the previous SOTA. The MobileBERT also showed a reasonable F1 but only outperformed Escalante et al. (2017) when bigger portions of the chats were given.
It should be noted that 

**Not very related:**
They used a figure in this paper, and I enjoyed it. The dots are the mean in different runs, and the highlighted areas represent the standard deviation.
![image](https://github.com/fani-lab/Osprey/assets/124281176/73f1e141-afd9-409a-b6d7-ea1be512f182)
